public/tabs.js

```diff
@@ -23,247 +14,280 @@
+        tabContent.setAttribute('tabindex', '0');
+        tabContent.setAttribute('aria-hidden', 'false');
+        tabContent.classList.add('active');
+
+        // Set focus when required
+        if (setFocus) {
+          tab.focus();
+        };
       };
-    };
-  };
 
-  // Deletes a tab and its panel
-  function deleteTab (event) {
-    var target = event.target;
-    var panel = document.getElementById(target.getAttribute('aria-controls'));
+      // Deactivate all tabs and tab panels
+      function deactivateTabs () {
+        for (let t = 0; t < tabs.length; t++) {
+          tabs[t].setAttribute('tabindex', '-1');
+          tabs[t].setAttribute('aria-selected', 'false');
+          tabs[t].classList.remove('active');
+          tabs[t].parentElement.classList.remove('active');
+          tabs[t].removeEventListener('focus', focusEventHandler);
+        }
 
-    target.parentElement.removeChild(target);
-    panel.parentElement.removeChild(panel);
-  };
+        for (let p = 0; p < panels.length; p++) {
+          panels[p].setAttribute('hidden', 'hidden');
+          panels[p].setAttribute('tabindex', '-1');
+          panels[p].setAttribute('aria-hidden', 'true');
+          panels[p].classList.remove('active');
+        }
+      };
 
-  // Determine whether there should be a delay
-  // when user navigates with the arrow keys
-  function determineDelay () {
-    var hasDelay = tablist.hasAttribute('data-delay');
-    var delay = 0;
-
-    if (hasDelay) {
-      var delayValue = tablist.getAttribute('data-delay');
-      if (delayValue) {
-        delay = delayValue;
-      }
-      else {
-        // If no value is specified, default to 300ms
-        delay = 300;
+      // Make a guess
+      function focusFirstTab () {
+        tabs[0].focus();
       };
-    };
 
-    return delay;
-  };
+      // Make a guess
+      function focusLastTab () {
+        tabs[tabs.length - 1].focus();
+      };
 
-  //
-  function focusEventHandler (event) {
-    var target = event.target;
+      // Detect if a tab is deletable
+      function determineDeletable (event) {
+        target = event.target;
 
-    setTimeout(checkTabFocus, delay, target);
-  };
+        if (target.getAttribute('data-deletable') !== null) {
+          // Delete target tab
+          deleteTab(event, target);
 
-  // Only activate tab on focus if it still has focus after the delay
-  function checkTabFocus (target) {
-    focused = document.activeElement;
+          // Update arrays related to tabs widget
+          generateArrays();
 
-    if (target === focused) {
-      activateTab(target, false);
-    };
-  };
+          // Activate the closest tab to the one that was just deleted
+          if (target.index - 1 < 0) {
+            activateTab(tabs[0]);
+          }
+          else {
+            activateTab(tabs[target.index - 1]);
+          };
+        };
+      };
+
+      // Deletes a tab and its panel
+      function deleteTab (event) {
```

Comment:
Under what circumstances do we delete a tab?

Reply:
A user can opt to delete  tab so it is not in the navigation. I am not sure if we even use this feature as it requires data-deletable so this can be removed